syntax = "proto3";

package api;

import "google/protobuf/timestamp.proto";

service API {
  rpc Ping(PingReq) returns (PingRes) {
    // Pings the server for updates and basic user info
  }

  rpc GetUser(GetUserReq) returns (User) {
    // Get info about a particular user
  }
}

message PingReq {}

message PingRes {
  // Basic info about user
  int64 user_id = 1;
  string username = 2;
  string name = 3;

  // Updates
  int64 unread_message_count = 4;
}

message User {
  string username = 1;
  string name = 2;
  string city = 3;
  double verification = 4;
  double community_standing = 5;
  uint32 num_references = 6;
  string gender = 7;
  uint32 age = 8;
  google.protobuf.Timestamp joined = 9; // not exact
  google.protobuf.Timestamp last_active = 10; // not exact
  string occupation = 11;
  string about_me = 12;
  string about_place = 13;
  repeated string languages = 14;
  repeated string countries_visited = 15;
  repeated string countries_lived = 16;
}

message GetUserReq {
  string user = 1;
}
